app/Http/Controllers/Api/v1/User/Controllers/GenerateTokenController.php

```diff
@@ -42,13 +44,42 @@ public function __invoke(Request $request): JsonResponse
         try {
             $user = $this->userLookup->getUpdatedUser($user, $azureUser->principal_name);
         } catch (ClientExceptionInterface) {
+            $log = new Log();
+            $log->user_id = $user->id;
+            $log->severity = '5';
+            $log->category = 'mensa';
+            $log->text = json_encode((object) [
+                'result' => "failure",
+                'data' => "Updating user failed: HTTP_BAD_GATEWAY"
+            ]);             
+            $log->text = json_encode();
+            $user->Log()->save($log);
+
             abort(Response::HTTP_BAD_GATEWAY);
         }
 
         if ($user == null) {
+            $log = new Log();
+            $log->user_id = $this->systemUser->id;
+            $log->severity = '5';
+            $log->category = 'mensa';
```

Comment:
Lots of duplicate code here, isn't it just user_id being different here? (and maybe the `text`)